vendor/bat-native-confirmations/src/bat/confirmations/internal/confirmations_impl.cc

```diff
@@ -302,43 +311,83 @@ bool ConfirmationsImpl::GetTransactionHistoryFromDictionary(
   for (auto& transaction_value : transactions_list_value) {
     base::DictionaryValue* transaction_dictionary;
     if (!transaction_value.GetAsDictionary(&transaction_dictionary)) {
-      return false;
+      DCHECK(false) << "Transaction should be a dictionary";
+      continue;
```

Comment:
It's OK to continue here when the data isn't as expected?

Reply:
yes, if transaction gets into a bad state, we should still keep good transactions

Reply:
OK, got it.

Reply:
Also this way bad transactions are still persisted in the state to help diagnose issues